src/Native/LdaNative/alias_multinomial_rng_int.hpp

```diff
@@ -108,7 +108,12 @@ namespace wood
             int32_t H_head = 0;
             int32_t H_tail = 0;
 
-            for (auto i = 0; i < proportion_int_.size(); ++i)
+
+            // note that i must fit into int32_t of L_[L_tail].first
+            if (static_cast<uint32_t>(std::numeric_limits<int32_t>::max()) < proportion_int_.size() )
```

Comment:
>int32_t [](start = 58, length = 7)

shouldnt this be uint32_t? Then you could remove the static_cast. Also why not have if (proportion_int_.size() > std::numeric_limits<uint32_t>::max())?

Reply:
oh I see why you are using the int32_t, ok so you are getting the max of int32_t, then cast that to uint32_t for the size comparison?

---
In reply to: [276882246](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276882246) [](ancestors = 276882246)